### PR TITLE
ffi: expose room membership through the RoomListItem and allow invited rooms to be build differently than "full" ones

### PR DESCRIPTION
**Pull request by stefanceriu**
We're finding ourselves in the situation in which we can't interact with invites through normal Room APIs as `full_room`s can't be build from the RoomListItem. Full rooms require the timeline to be configured before use and the timeline can't be configured because encryption cannot be fetched for invited rooms on homeservers that have previews disabled (see #3848 and #3850)

In response we now expose the room's membership directly from the `RoomListItem` so that the final client can chose which of the 2 rooms types (invited or full) to ask for before using aforementioned APIs.

Powers https://github.com/element-hq/element-x-ios/pull/3189

### CODE REVIEW

File: bindings/matrix-sdk-ffi/src/room_list.rs [hunk 1]
```diff
@@ -603,10 +605,35 @@ impl RoomListItem {
         Ok(RoomInfo::new(self.inner.inner_room()).await?)
     }
 
+    /// The room's current membership state
```
Comment (bnjbvr) [inline bindings/matrix-sdk-ffi/src/room_list.rs:608]:
```suggestion
    /// The room's current membership state.
```

File: bindings/matrix-sdk-ffi/src/room_list.rs [hunk 2]
```diff
@@ -50,6 +50,8 @@ pub enum RoomListError {
     InitializingTimeline { error: String },
     #[error("Event cache ran into an error: {error}")]
     EventCache { error: String },
+    #[error("The requested room doesn't match the membership requirements")]
+    IncorrectRoomMembership,
```
Comment (bnjbvr) [inline bindings/matrix-sdk-ffi/src/room_list.rs:54]: bonus points if you include two fields `expected` / `observed` that help matching the requirements, but not mandatory for merging :)

**Review by bnjbvr — APPROVED** [review]
LGTM, thanks!

File: bindings/matrix-sdk-ffi/src/room_list.rs (hunk 2 above)
Comment (stefanceriu) [reply bindings/matrix-sdk-ffi/src/room_list.rs:54]: Noice, bonus points for me it is then! 😁

**Pull request merged**